contracts/payloads/IGP32/PayloadIGP32.sol

```diff
@@ -0,0 +1,563 @@
+            "msg.sender-not-allowed"
+        );
+
+        uint256 totalActions = 1;
+        address[] memory targets = new address[](totalActions);
+        uint256[] memory values = new uint256[](totalActions);
+        string[] memory signatures = new string[](totalActions);
+        bytes[] memory calldatas = new bytes[](totalActions);
+
+        // Action 1: call executePayload on timelock contract to execute payload related to Fluid
+        targets[0] = address(TIMELOCK);
+        values[0] = 0;
+        signatures[0] = "executePayload(address,string,bytes)";
+        calldatas[0] = abi.encode(ADDRESS_THIS, "execute()", abi.encode());
+
+        uint256 proposedId = GOVERNOR.propose(
+            targets,
+            values,
+            signatures,
+            calldatas,
+            description
+        );
+
+        require(proposedId == PROPOSAL_ID, "PROPOSAL_IS_NOT_SAME");
+    }
+
+    function execute() external {
+        require(address(this) == address(TIMELOCK), "not-valid-caller");
+
+        // Action 1: Deploy wBTC/ETH and ETH/wBTC vaults.
+        action1();
+
+        // Action 2: Approve fUSDC and fUSDT protocols to spend the reserves tokens
+        action2();
+
+        // Action 3: Dust reserves tokens allowance
+        action2();
+    }
+
+    function verifyProposal() external view {}
+
+    /***********************************|
+    |     Proposal Payload Actions      |
+    |__________________________________*/
+
+    /// @notice Action 1: Update rewards for fUSDC & fUSDT.
+    function action1() internal {
+        IFTokenAdmin(F_USDC).updateRewards(
+            0xD8CA3D94f2EF112e5232a3805dc17dc79D390603
+        );
+
+        IFTokenAdmin(F_USDT).updateRewards(
+            0xD8CA3D94f2EF112e5232a3805dc17dc79D390603
+        );
+    }
+
+    /// @notice Action 2: Approve fUSDC and fUSDT protocols to spend the reserves tokens
+    function action2() internal {
+        address[] memory protocols = new address[](2);
+        address[] memory tokens = new address[](2);
+        uint256[] memory amounts = new uint256[](2);
+
+        // fUSDC
+        protocols[0] = F_USDC;
+        tokens[0] = USDC_ADDRESS;
+        amounts[0] = 250_000 * 1e6; // 250k USDC
+
+        // fUSDT
+        protocols[1] = F_USDT;
+        tokens[1] = USDT_ADDRESS;
+        amounts[1] = 250_000 * 1e6; // 250k USDT
+
+        FLUID_RESERVE.approve(protocols, tokens, amounts);
+    }
+
+    /// @notice Action 3: Dust reserves tokens allowance
+    function action3() internal {
+        address[] memory protocols = new address[](2);
+        address[] memory tokens = new address[](2);
+        uint256[] memory amounts = new uint256[](2);
+
+        protocols[0] = 0x6F72895Cf6904489Bcd862c941c3D02a3eE4f03e;
+        tokens[0] = wBTC_ADDRESS;
+        amounts[0] = 0.8 * 1e8;
+
+        protocols[1] = 0x3A0b7c8840D74D39552EF53F586dD8c3d1234C40;
+        tokens[1] = wBTC_ADDRESS;
+        amounts[1] = 0.8 * 1e8;
+
```

Comment:
can we also add borrow token dust amounts?
```suggestion
        protocols[2] = 0x6F72895Cf6904489Bcd862c941c3D02a3eE4f03e;
        tokens[2] = USDC_ADDRESS;
        amounts[2] = 100 * 1e6;
                
        protocols[3] = 0x3A0b7c8840D74D39552EF53F586dD8c3d1234C40;
        tokens[3] = USDT_ADDRESS;
        amounts[3] = 100 * 1e6;
```

borrow token should not be needed to be transferred in when doing rebalance() but just to be safe and guarantee filling WBTC rewards will work as we anyway usually give dust allowance